examples/ENS/subdomainBundle.en.js

```diff
@@ -11,8 +11,9 @@ class Token {
         let ensName = this.props.ensName;
         let subdomainElement = document.getElementById("subdomain");
         subdomainElement.addEventListener("keyup", function() {
-            let subdomain = subdomainElement.value.toString().trim().toLowerCase().replace(ensName, "");
-            document.getElementById("subdomainHash").innerHTML = namehash.hash(subdomain + "." + ensName);
+            let subdomain = subdomainElement.value.toString().trim().toLowerCase();
+            document.getElementById("subdomainHash").innerHTML =
+                namehash.hash(subdomain.substring(0, subdomain.indexOf(".")) + "." + ensName);
```

Comment:
This line doesn't work correctly if `subdomain` doesn't contain a ".". Maybe be more strict about the value entered subdomain?

There are just 2 (or 3) possible types:

A) "name.${ensName}"
B) "name"
C) can "name" contain "."? eg. "my.sub.joe.eth"?

We can decide if we only want to support either one or all of them, and it'll be easier to code.

Reply:
Support both

Reply:
Then how about a more specific if-else [1]. How about:

```
if subdomain.suffix == ".${ensName}" {
  //substring with the length of ".${ensName}"
} else {
  //depending on whether (C) above is possible and act accordingly
}
```

[1] `indexOf()` returns -1 if it's not found

Reply:
Alternative we can cheat a little and display the text box like:

Subdomain: [          ].joe.eth

And expect the user to never type (A)

Reply:
Hmm, that might be a bit more friendly